tests/acceptance/bootstrap/SettingsContext.php

```diff
@@ -596,11 +597,109 @@ public function getBodyForNotificationSetting(string $user): array {
 						"resource" => [
 							"type" => $settingsValue->resource->type
 						],
-						"boolValue" => true
 					]
 				];
 			}
 		}
-		throw new Exception(('`disable-email-notifications` not found in the setting list'));
+		throw new Exception(("'$event' not found in the setting list"));
+	}
+
+	/**
+	 * @When /^user "([^"]*)" (disables|enables) notification for following event using Settings API:$/
+	 *
+	 * @param string $user
+	 * @param string $enableOrDisable
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function userDisablesNotificationForFollowingEventUsingSettingsApi(
+		string $user,
+		string $enableOrDisable,
+		TableNode $table
+	): void {
+		$settings = $table->getRowsHash();
+		foreach ($settings as $event => $value) {
+			$body = $this->getBodyForNotificationSetting($user, $event);
+			if (str_contains($value, "mail")) {
+				$body["value"]["collectionValue"]["values"][]
+					= ["key" => "mail","boolValue" => $enableOrDisable === "enables"];
+			}
+			if (str_contains($value, "in-app")) {
+				$body["value"]["collectionValue"]["values"][]
+					= ["key" => "in-app","boolValue" => $enableOrDisable === "enables"];
+			}
+			$response = HttpRequestHelper::sendRequest(
+				$this->featureContext->getBaseUrl() . "/api/v0/settings/values-save",
+				$this->featureContext->getStepLineRef(),
+				'POST',
+				$this->featureContext->getActualUsername($user),
+				$this->featureContext->getPasswordForUser($user),
+				null,
+				json_encode($body),
+			);
+			$this->featureContext->setResponse($response);
+		}
+	}
+
+	/**
+	 * @Given /^user "([^"]*)" has (disabled|enabled) notification for following event using Settings API:$/
+	 *
+	 * @param string $user
+	 * @param string $enableOrDisable
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function userHasDisabledOrEnabledNotificationForFollowingEventUsingSettingsApi(
+		string $user,
+		string $enableOrDisable,
+		TableNode $table
+	): void {
+		$settings = $table->getRowsHash();
+		foreach ($settings as $event => $value) {
+			// var_dump("$event: $value");
+			$body = $this->getBodyForNotificationSetting($user, $event);
+			if (str_contains($value, "mail")) {
+				$body["value"]["collectionValue"]["values"][]
+					= ["key" => "mail","boolValue" => $enableOrDisable === "enables"];
+			}
+			if (str_contains($value, "in-app")) {
+				$body["value"]["collectionValue"]["values"][]
+					= ["key" => "in-app","boolValue" => $enableOrDisable === "enables"];
+			}
+			$response = HttpRequestHelper::sendRequest(
```

Comment:
now a days i am seeing less use of `HttpRequestHelper::post` @saw-jan what you think on this topic